kserve-module/tests/e2e/conftest.py

```
@@ -0,0 +1,189 @@
+def is_cr_ready(cr):
+    """Check if a Kserve CR dict has Ready=True."""
+    conditions = cr.get("status", {}).get("conditions", [])
+    return any(c.get("type") == "Ready" and c.get("status") == "True" for c in conditions)
+
+
+# ---------------------------------------------------------------------------
+# Helper functions — shell / kubectl
+# ---------------------------------------------------------------------------
+def run(cmd, check=True, timeout=60):
+    """Run a shell command and return the result."""
+    result = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)